src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -109,13 +109,13 @@ private void buildGeneralParameters() {
         context.put("build_time", formattedTime);
 
         // build number is not mandatory
-        String buildNumber = configuration.getBuildNumber();
-        if (StringUtils.isNotBlank(buildNumber) && configuration.isRunWithJenkins()) {
-            if (NumberUtils.isCreatable(buildNumber)) {
-                context.put("build_previous_number", Integer.parseInt(buildNumber) - 1);
-            } else {
-                LOG.log(Level.INFO, String.format("Could not parse build number: %1$s.", configuration.getBuildNumber()));
-            }
+        // the build number could contain " ". calling strip will return 'null' to
```

Comment:
when this could heppen?

Reply:
After some doing some intense testing I know this condition would never happen.  It was in the original code however and I thought the developer knew something I didn't.  I also did a test where the Jenkins build # was overridden with a semantic version (this does happen where I work) as well as passing a blank string and a string with an empty space.  What learned was that although the BUILD_NUMBER global variable in Jenkins was the value the user defined the number that is passed to this method is always a whole #.  Would you like me to modify the method and submit a new pull request?

I also figured out what the cause of the original failure was in my case.  On the Jenkins master there are some custom jars in java.ext.dirs one of those jars had commons.lang3 as a dependency which inadvertently returned threw the exception.  That dependency wasn't being used in the jar in question, and once it was removed the method works as originally designed.

It's unclear to me as to why the condition throws an exception except to say that it's a CLASSPATH issue.  I am Java 5,6,7 certified but my current role at my place of employment isn't development.

Reply:
I don't want to keep something what is hard to understand and maintain.